DxDispatch/src/dxdispatch/DxModules.cpp

```diff
@@ -76,10 +67,31 @@ void* Module::GetSymbol(const char* name)
 #endif
 }
 
-D3d12Module::D3d12Module(const char* moduleName) : Module(moduleName)
+D3d12Module::D3d12Module(bool disableAgilitySDK, const char* moduleName) : Module(moduleName)
 {
     if (m_module)
     {
+#if !defined(_GAMING_XBOX) && defined(WIN32)
+        if(!disableAgilitySDK)
+        {
+            InitSymbol(&m_d3d12SDKConfiguration, "D3D12GetInterface");
+            if(m_d3d12SDKConfiguration)
+            {
+                ComPtr<ID3D12SDKConfiguration1> pD3D12SDKConfiguration;
+                HRESULT hr = m_d3d12SDKConfiguration(CLSID_D3D12SDKConfiguration, IID_PPV_ARGS(&pD3D12SDKConfiguration));
+                ComPtr<ID3D12DeviceFactory> deviceFactory;
+                if(SUCCEEDED(hr))
+                {
+                    hr = pD3D12SDKConfiguration->CreateDeviceFactory(D3D12SDKVersion, D3D12SDKPath, IID_PPV_ARGS(&deviceFactory));
```

Comment:
Assuming this can fail, it seems best to throw an exception since the app is no longer respecting the user's wishes.

Reply:
It did not fail before so I was keeping existing behavior.   I am happy to throw an exception though.